File: rotational-cipher/RotationalCipher.fs
module RotationalCipher
open System

let rotate shiftKey text = 
    let rotateletter shiftKey c = 
        match int c + shiftKey with
        | r when Char.IsUpper c && r > int 'Z' -> r - 26 |> char
        | r when Char.IsLower c && r > int 'z' -> r - 26 |> char
        | r -> int r |> char
    let mapper (c:char) = if Char.IsLetter c then rotateletter shiftKey c else c
    text |> Seq.map mapper |> String.Concat

File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Linq;

public static class RotationalCipher
{
    private static readonly (int, int) lower = ((int)'a', (int)'z');
    private static readonly (int, int) upper = ((int)'A', (int)'Z');

    public static string Rotate(string text, int shiftKey)
        => string.Concat(text.Select(o => char.IsLetter(o) ? Shift(o, shiftKey) : o));

    private static char Shift(char source, int shiftKey)
    {
        var (min, max) = char.IsLower(source) ? lower : upper;
        var newVal = (int)source + shiftKey;
        return (char)(newVal > max ? min + (newVal - max - 1) : newVal);
    }   
}